src/Core/Managed/Shared/Extensibility/Metrics/DependencyMetricExtractor.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using System.Collections.Generic;
+
+namespace Microsoft.ApplicationInsights.Extensibility.Metrics
+{
+    public class DependencyMetricExtractor : MetricExtractorTelemetryProcessorBase
```

Comment:
@vitalyf007 @dnduffy I have actually considered both approached form the start. I picked this because I wanted to follow guidance by @SergeyKanzhelev to make things "plugable as if I was a customer". So a customer should be able to create a metric extractor and to plug it in using a familiar paradigm. This means that we have two approaches:
- The one I used.
- A similar, but from the ExtractorBase I would derive a single "Auto-Collected Metric Extractor" that would do all *our* extraction and leave the users the possibility to create more extractors.
(- on contrary, a processor that contains all extractors would require the user to configure things differently from everything else)
Please let me know if you prefer me to go into more detail about who I chose the first of the two.